x-pack/plugins/alerting/common/rule.ts

```diff
@@ -121,17 +121,27 @@ export const RuleActionTypes = {
 
 export type RuleActionTypes = typeof RuleActionTypes[keyof typeof RuleActionTypes];
 
-export interface RuleAction {
+export interface RuleDefaultAction {
   uuid?: string;
   group: string;
   id: string;
   actionTypeId: string;
   params: RuleActionParams;
   frequency?: RuleActionFrequency;
   alertsFilter?: AlertsFilter;
-  type?: typeof RuleActionTypes.DEFAULT;
+  type: typeof RuleActionTypes.DEFAULT;
 }
 
+export interface RuleSystemAction {
+  uuid?: string;
```

Comment:
Why `uuid` is optional here? In which case it'll be or not to be.

Reply:
Because the APIs have the `uuid` optional for the default actions. The rule client adds it when needed. We can make it required. In this case, the APIs should add it before passing it to the rules client.